### PR TITLE
HDS-2275 Login pattern page and documentation

### PR DESCRIPTION
**Pull request by nooranen**
## Description

PR for login documentation changes (WIP)

**To do:**

Header component docs pages
- [x] Usage
- [x]  Code
- [ ] accessibility
- [ ] Customisation

- [x] Login component docs - what is needed to link back to Header.Login & pattern page 

- [x] Correct demos & technical information (for login pattern page & header component page)

- [ ] Login & Profile pattern page

## Add to changelog
- [x] Added needed line to changelog

### CODE REVIEW

File: site/src/docs/components/header/index.mdx [hunk 1]
```diff
@@ -303,8 +323,18 @@ The Header.Search component is to be used for whole site wide search. Currently
 The Header.ActionBarItem is a generic component for displaying icon buttons on the action bar, optionally accompanied with dropdowns.
 
 - The Header.ActionBarItem extends native button element's attributes.
-- The item can be positioned behind a dividing line, setting it apart from the rest of the items with the prop `fixedRightPosition`. This is mainly reserved for log in button.
-- Please note that under the S breakpoint (576px) the ActionBarItems don't fit in the screen well and hiding them should be considered. We are working on a solution for this.
+- The item can be positioned behind a dividing line, setting it apart from the rest of the items with the prop `fixedRightPosition`. This is mainly reserved for [Header.Login](#with-login).
+  - From breakpoint M (≥ 768px) and down Header.ActionBarItem using the `fixedRightPosition` prop is hidden and the contents are moved inside the [the Header.ActionBar component's](#with-action-bar) menu alongside NavigationMenu.
+    - No more than two ActionBarItems are recommended to use this high priority position, to conserve mobile menu's usability.
```
Comment (laurakarhu) [inline site/src/docs/components/header/index.mdx:328]: It is higly recommended to have maximum 2 ActionBarItems...

**Review by laurakarhu — COMMENTED** [review]
Nice job on content!

**Review by Riippi — APPROVED** [review]
👍

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## [3.X.X] - Month, XX, 202X
+## [3.9.0] - June, X, 2024
```
Comment (mrTuomoK) [inline CHANGELOG.md:8]: Eh... you mean July? 😄

File: site/src/docs/patterns/login/building-login.mdx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+#### Requirements
+
+The following steps are required for user authentication:
+
+- The url of the service connecting to the OIDC server <InternalLink href="/components/login/usage/#loginprovider">must be registered with a scope and a client ID</InternalLink>.
+- The callback from the OIDC server should be handled with the <InternalLink href="/components/login/usage/#logincallbackhandler">LoginCallbackHandler</InternalLink> component.
+
+#### Log in process
+
+Logging in or out of an OIDC server requires browser redirections to the server and back. UI's only show login and logout buttons. No other form elements are needed.
```
Comment (laurakarhu) [inline site/src/docs/patterns/login/building-login.mdx:55]: Better readability if we write it open. The user interface only shos login and logout buttons.

File: site/src/docs/patterns/login/building-login.mdx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+The `login` call is asyncronous and may fail if the OIDC server is offline or the service URL is not registered. UI's must indicate loading has started, and possible errors must be handled by the UI.
```
Comment (laurakarhu) [inline site/src/docs/patterns/login/building-login.mdx:57]: The user interface must indicate that the loading has started

File: site/src/docs/patterns/login/building-login.mdx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+The <InternalLink href="/patterns/login/building-login/#header-login-button">HDS Header Login button</InternalLink> handles all of these scenarios.
+
+#### Log out process
+
+Logging out is almost identical to the log-in process. It is also asyncronous and may fail.
```
Comment (laurakarhu) [inline site/src/docs/patterns/login/building-login.mdx:63]: Log out is almost identical to the log in process.

File: site/src/docs/patterns/login/building-login.mdx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+The <InternalLink href="/patterns/login/building-login/#header-logout-button">HDS Header Logout button</InternalLink> handles all of these scenarios.
+
+#### User data from the OIDC server
+
+User information is stored in tokens which are decrypted by the Login system and an user object is provided to UI's. It holds, for example, user's name and email address.
```
Comment (laurakarhu) [inline site/src/docs/patterns/login/building-login.mdx:69]: is provided to user interfaces.

File: site/src/docs/patterns/login/building-login.mdx [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+User object can be accessed with the <InternalLink href="/components/login/usage/#useauthenticateduser">useAuthenticatedUser</InternalLink> hook.
+
+The <InternalLink href="/patterns/login/building-login/#header-user-menu">HDS Header User menu</InternalLink> automatically shows the user's name and initials.
+
+#### User data from the Profile Service
+
+<ExternalLink href="https://profiili.hel.fi">Profile information</ExternalLink> is not used by the HDS. It can be fetched
+with the GraphQL module, but it is not used in HDS components. The user object returned by the OIDC server has a `profile`
```
Comment (laurakarhu) [inline site/src/docs/patterns/login/building-login.mdx:78]: Link to the GraphlQL module?

**Review by laurakarhu — CHANGES_REQUESTED** [review]
Minor improvements to the readability and some typos

**Review by laurakarhu — APPROVED** [review]
These minor tweaks don't prevent release, but should be done in proper moment

**Pull request merged**